libethereum/Account.h

```diff
@@ -152,29 +140,24 @@ class Account
 	/// database.
 	void setStorageCache(u256 _p, u256 _v) const { const_cast<decltype(m_storageOverlay)&>(m_storageOverlay)[_p] = _v; }
 
-	/// @returns true if we are in the contract-conception state and setCode is valid to call.
-	bool isFreshCode() const { return m_codeHash == c_contractConceptionCodeHash; }
+	/// @returns the hash of the account's code.
+	h256 codeHash() const { return m_codeHash; }
 
-	/// @returns true if we are either in the contract-conception state or if the account's code is not
-	/// empty.
-	bool codeBearing() const { return m_codeHash != EmptySHA3; }
-
-	/// @returns the hash of the account's code. Must only be called when isFreshCode() returns false.
-	h256 codeHash() const { assert(!isFreshCode()); return m_codeHash; }
+	bool isFreshCode() const { return m_hasNewCode; }
```

Comment:
Previously, this function returned true even during creation. Now it only returns true after code has been deployed.